src/pages/galleries/index.vue

```diff
@@ -48,41 +67,86 @@
       PaginationNavInput: () => import('../../components/generic/PaginationNavInput')
     },
     middleware: 'sanitisePageQuery',
-    asyncData({ query, error, app, store }) {
-      const variables = {
-        locale: app.i18n.isoLocale(),
-        preview: query.mode === 'preview',
-        limit: PER_PAGE,
-        skip: (store.state.sanitised.page - 1) * PER_PAGE
-      };
-
-      return app.$contentful.query('galleryFoyerPage', variables)
-        .then(response => response.data.data)
-        .then(data => {
-          return {
-            galleries: data.imageGalleryCollection.items,
-            total: data.imageGalleryCollection.total,
-            page: store.state.sanitised.page,
-            perPage: PER_PAGE
-          };
-        })
-        .catch((e) => {
-          error({ statusCode: 500, message: e.toString() });
-        });
-    },
     data() {
       return {
+        galleries: [],
         perPage: PER_PAGE,
-        page: null
+        total: 0
       };
     },
+    async fetch() {
+      if (this.setGalleriesEnabled) {
+        await this.fetchSetGalleries();
+      } else {
+        await this.fetchContentfulGalleries();
+      }
+    },
     head() {
       return {
         title: this.$pageHeadTitle(this.$tc('galleries.galleries', 2))
       };
     },
-    watchQuery: ['page'],
+    computed: {
+      setGalleriesEnabled() {
+        return this.$features.setGalleries;
+      },
+      page() {
+        return Number(this.$route.query.page || 1);
+      }
+    },
+    watch: {
+      '$route.query.page': '$fetch'
+    },
+    watchQuery: ['page'], // TODO: remove when using set galleries
```

Comment:
Is this still actually doing anything? I think not with the switch to using the fetch hook.

Reply:
no, left over from a previous iteration when it was needed.